sihl/src/web_bearer_token.ml

```diff
@@ -2,28 +2,48 @@ let key : string Opium.Context.key =
   Opium.Context.Key.create ("token", Sexplib.Std.(sexp_of_string))
 ;;
 
-let find req = Opium.Context.find_exn key req.Opium.Request.env
-let find_opt req = Opium.Context.find key req.Opium.Request.env
+exception Token_not_found
+
+let find req =
+  try Opium.Context.find_exn key req.Opium.Request.env with
+  | _ ->
+    Logs.err (fun m -> m "No bearer token found");
+    Logs.info (fun m ->
+        m "Have you applied the bearer token middleware for this route?");
+    raise @@ Token_not_found
+;;
+
+let find_opt req =
+  try Some (find req) with
+  | _ -> None
+;;
 
 let set token req =
   let env = req.Opium.Request.env in
   let env = Opium.Context.add key token env in
   { req with env }
 ;;
 
-let middleware =
+let default_handler _ =
+  Opium.Response.of_plain_text "Unauthorized 401"
```

Comment:
I assumed that always comes from the web server, maybe some browsers have a sane default though. Let's play it safe, otherwise we have to send empty string as body.

Reply:
You're right, let's also add it to other spots, like default unauthorized handler in CSRF middleware.